### PR TITLE
add zip test for blob with negative size

### PR DESCRIPTION
**Pull request by wfurt**
verifies we throw on malformed zip entry.

### CODE REVIEW

File: src/libraries/System.IO.Compression/tests/ZipArchive/zip_InvalidParametersAndStrangeFiles.cs [hunk 1]
```diff
@@ -2399,5 +2399,187 @@ public static async Task OpenWithFileAccess_DisposedArchive_Throws(bool async)
...
+                while ((read = s.Read(buffer, 0, buffer.Length)) > 0)
+                {
+                    totalRead += read;
+                    // Guard against the negative size being misinterpreted as a huge unsigned
+                    // length that would let the read loop run forever.
+                    Assert.True(totalRead <= 1024 * 1024, "Read returned more data than the archive contains.");
+                }
+                Assert.Equal(0, totalRead);
+            }
+            catch (Exception ex) when (ex is InvalidDataException or ArgumentOutOfRangeException)
```
Comment (rzikm) [inline src/libraries/System.IO.Compression/tests/ZipArchive/zip_InvalidParametersAndStrangeFiles.cs]: We should not expect ArgumentOutOfRangeException here

**Review by rzikm — APPROVED** [review]
LGTM, modulo comments

**Comment by rzikm** [comment]
@wfurt Do you want to continue with this PR? it is slowly getting stale

File: src/libraries/System.IO.Compression/tests/ZipArchive/zip_InvalidParametersAndStrangeFiles.cs [hunk 2]
```diff
@@ -2434,5 +2434,182 @@ public static async Task OpenWithFileAccess_DisposedArchive_Throws(bool async)
...
+            // harmful behavior. The archive should continue to handle the entry safely.
+            byte[] zipArchive = CreateZipWithNegativeZip64UncompressedSize();
+
+            using ZipArchive archive = new ZipArchive(new MemoryStream(zipArchive), ZipArchiveMode.Read);
+            Assert.Equal(1, archive.Entries.Count);
+            ZipArchiveEntry entry = archive.Entries[0];
+
+            // The negative value propagates as-is to the entry's Length property.
+            // This is observable but not by itself harmful.
+            Assert.Equal(-1L, entry.Length);
```
Comment (rzikm) [inline src/libraries/System.IO.Compression/tests/ZipArchive/zip_InvalidParametersAndStrangeFiles.cs:2453]: I don't think this is the behavior we want, we should not return garbage data just so that we would throw later in Open(). We should validate the length ASAP in the process, which likely means during initial central directory parsing

Comment (alinpahontu2912) [reply src/libraries/System.IO.Compression/tests/ZipArchive/zip_InvalidParametersAndStrangeFiles.cs:2453]: I agree, I believe the checks can be made inside the TryGetZip64BlockFromGenericExtraField(Async)

**Comment by alinpahontu2912** [comment]
Closed this because of follow-up here: https://github.com/dotnet/runtime/pull/129121
